config/main.py

```diff
@@ -290,6 +273,77 @@ def is_interface_bind_to_vrf(config_db, interface_name):
         return True
     return False
 
+# Validate whether a given namespace name is valid in the device.
+def validate_namespace(namespace):
+    if not sonic_device_util.is_multi_npu():
+        return True
+
+    namespaces = sonic_device_util.get_all_namespaces()
+    if namespace in namespaces['front_ns'] + namespaces['back_ns']:
+        return True
+    else:
+        return False
+
+# Get the table name based on the interface type
+def get_port_table_name(interface_name):
+    """Get table name by port_name prefix
+    """
+    if interface_name.startswith("Ethernet"):
```

Comment:
Can we accomplish this without assuming the interface name begins with "Ethernet" or "PortChannel", etc.? If we decide to change the port naming conventions in SONiC, this will break.

Reply:
Thanks Joe.  I have created this new API similar to an existing API **def get_interface_table_name(interface_name)** which returns back the INTERFACE table name. This new API was needed as I would need the PORT_TABLE names for my use case. 

But anyways the reference made to interface names are all SONIC standard names. Please confirm if that is ok.

Reply:
We have discussed the idea of changing the SONiC interface naming convention in the future. I feel like we should make our utilities agnostic of interface names. I'd like to get @lguohan's opinion here.

Reply:
Sure Joe,  An approach  I think is to define the interface name prefixes
ETHERNET_INTF_NAME_PREFIX = 'Ethernet'
ETHERNET_BACKPLANE_INTF_NAME_PREFIX = 'Ethernet-BP'
PORTCHANNEL_INTF_NAME_PREFIX = 'PortChannel'
VLAN_INTF_NAME_PREFIX = 'Vlan'
LOOPBACK_INTF_NAME_PREFIX = 'Loopback'

+ access utilities for prefix's in a common utility python script ( eg: sonic_device_util.py -- or -- a new file like sonic_interface.py ) and start using it in all places like config/show commands/minigraph parsing etc.
 
Another option will be to introduce a config file ( a json file)  for use which will have the mapping between the interface type viz. Ethernet/Ethernet-BP/PortChannel/Vlan/Loopback  -- to the actual sonic interface name prefix string.

@jleveque @lguohan  let me know your opinion , or any other suggestions.

Reply:
I have added an interface name prefix dictionary here for the various SONIC interface name prefixes. I have kept it here for now, but if this approach is ok for you -- I will move this definition to a common place which could be used by scripts/utilities. We can provide helper utilities to access this dictionary as well.

@jleveque @lguohan, please take a look and let me know if this approach is ok

Reply:
@lguohan to comment on this.

Reply:
@lguohan please share your comments on this.

I have one more suggestion, **Can I create a follow up item/issue to cleanup this and existing code in this file for "interface name" direct references. Replace them with interface related API's which we can introduce in a new file eg: sonic_interface.py in the sonic-py-common package ?**